### PR TITLE
Load newly-editable libraries dynamically in Sprite Lab

### PR DESCRIPTION
**Pull request by joshlory**
Depends on PR https://github.com/code-dot-org/code-dot-org/pull/23821.

### CODE REVIEW

**Comment by joshlory** [comment]
[Ability to edit helper libraries on levelbuilder](https://trello.com/c/o3XxUjPc/225-ability-to-edit-helper-libraries-on-levelbuilder)

**Comment by joshlory** [comment]
Yep!  For example:

https://github.com/code-dot-org/code-dot-org/blob/9a7ee7c94e48756157b19d36b023a66b97fdc307/dashboard/config/scripts/levels/New%20Sprite%20Lab%20Project.level#L41-L43

File: dashboard/app/controllers/libraries_controller.rb [hunk 1]
```diff
@@ -1,6 +1,10 @@
 class LibrariesController < ApplicationController
-  before_action :require_levelbuilder_mode
-  load_and_authorize_resource find_by: :name
+  before_action :require_levelbuilder_mode, except: :show
+  load_and_authorize_resource find_by: :name, except: :show
```
Comment (balderdash) [inline dashboard/app/controllers/libraries_controller.rb]: I think everyone is granted permission to read libraries in abilities.rb, so we shouldn't need to skip validation: https://github.com/code-dot-org/code-dot-org/blob/3d41cd121d696d359d79837c7a5550676f9f93fe/dashboard/app/models/ability.rb#L10

Comment (joshlory) [reply dashboard/app/controllers/libraries_controller.rb]: In this case I think I still need it, to skip the "load" part, so I can manually load from the cache.

**Comment by joshlory** [comment]
@balderdash PTAL!

File: apps/src/gamelab/GameLab.js [hunk 2]
```diff
@@ -1208,6 +1202,38 @@ GameLab.prototype.onP5Preload = function () {
...
+let libraryPreload;
+GameLab.prototype.loadLibraries_ = function () {
+  if (!libraryPreload) {
+    this.loadValidationCodeIfNeeded_();
+    libraryPreload = Promise.all(this.level.helperLibraries.map(this.loadLibrary_.bind(this)));
+  }
+  return libraryPreload;
+};
+
+GameLab.prototype.loadLibrary_ = function (name) {
```
Comment (balderdash) [inline apps/src/gamelab/GameLab.js:1233]: This looks like a nice place for `async`:
```
GameLab.prototype.loadLibrary_ = async function (name) {
  if (this.libraries[name]) {
    return;
  }
  const response = await fetch('/libraries/' + name);
  this.libraries[name] = await response.text();
};
```

File: apps/test/unit/gamelab/GameLabJrTest.js [hunk 3]
```diff
@@ -48,16 +47,7 @@ describe('Game Lab Jr Helper Library', () => {
...
-    // Declarations within eval() aren't applied to the calling scope in
-    // strict mode. Replace global variable/function declarations with explicit
-    // window property declarations.
-    const lib = GameLabJrLib
-        .replace(/}function (\w+)/gm, '};window.$1 = function ')
-        .replace(/\bfunction (\w+)/gm, 'window.$1 = function ')
-        .replace(/\bvar /gm, 'window.');
-
-    eval(lib); // eslint-disable-line no-eval
+    require('!!script-loader!@cdo/interpreted/GameLabJr.interpreted.js');
```
Comment (balderdash) [inline apps/test/unit/gamelab/GameLabJrTest.js:50]: :tada:

File: apps/src/gamelab/GameLab.js (hunk 2 above)
Comment (joshlory) [reply apps/src/gamelab/GameLab.js:1233]: https://caniuse.com/#feat=fetch not available in IE11?  Does it get polyfilled by our build process?

Comment (balderdash) [reply apps/src/gamelab/GameLab.js:1233]: Yep, we include a fetch polyfill in all our entry points here: https://github.com/code-dot-org/code-dot-org/blob/151f174856ceb4feef47af945cbde25214cb0219/apps/Gruntfile.js#L559

**Pull request merged**